app/controllers/application_controller.rb

```diff
@@ -41,4 +42,17 @@ def disable_site_header!
   def self.disable_site_header!(*args)
     before_action :disable_site_header!, *args
   end
+
+  private
+  def storable_location?
+    request.get? && is_navigational_format? && !devise_controller? && !request.xhr?
```

Comment:
We don't need to store this for logged in users. So you can add `&& !user_signed_in?`